pennylane/labs/resource_estimation/resource_container.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Base classes for resource estimation."""
+import copy
+from collections import defaultdict
+from dataclasses import dataclass, field
+
+from pennylane.labs.resource_estimation import ResourceOperator
+
+
+class CompressedResourceOp:
+    r"""Instantiate the light weight class corressponding to the operator type and parameters.
+
+    Args:
+        op_type: the class object for of an operation which inherits from '~.ResourceOperator'
+        params (dict): a dictionary containing the minimal pairs of parameter names and values
+                    required to compute the resources for the given operator!
+
+    .. details::
+
+        This representation is the minimal amount of information required to estimate resources for the operator.
+
+        **Example**
+
+        >>> op_tp = CompressedResourceOp(ResourceHadamard, {"num_wires":1})
```

Comment:
Yes, it should work fine. There are no checks to validate the inputs. I think its fine to keep it like this because we don't want users to construct these CompressedResourceOp objects anyways. Its more for us to have an efficient internal representation for the object.